lib/util/update-metadata-builder.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Utility for building the metadata patch request
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var assert = require('assert');
+
+/**
+ * Utility for building the metadata patch request for files and folders
+ *
+ * @constructor
+ * @returns {void}
+ */
+function UpdateMetadataBuilder() {
+	this.body = [];
+}
+
+/**
+ * Insert add operation to the request body.
+ *
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertAdd = function(path, value) {
```

Comment:
Since all these begin with 'insert', maybe it's easier to just name them after the operation, e.g. "add" or "replace"